Content.Server/Weapon/Melee/EnergySword/EnergySwordSystem.cs

```diff
@@ -0,0 +1,146 @@
+using System.Collections.Generic;
+using Content.Shared.Item;
+using Content.Server.Tools.Components;
+using Content.Shared.Interaction;
+using Content.Shared.ActionBlocker;
+using Content.Shared.FixedPoint;
+using Robust.Server.GameObjects;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Audio;
+using Robust.Shared.Player;
+using Robust.Shared.IoC;
+using Robust.Shared.Maths;
+using Robust.Shared.Random;
+using System;
+
+namespace Content.Server.Weapon.Melee.Esword
+{
+    internal class EnergySwordSystem : EntitySystem
+    {
+        [Dependency] private readonly ActionBlockerSystem _blockerSystem = default!;
+        [Dependency] private readonly IRobustRandom _random = default!;
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            SubscribeLocalEvent<EnergySwordComponent, ComponentInit>(OnComponentInit);
+            SubscribeLocalEvent<EnergySwordComponent, MeleeHitEvent>(OnMeleeHit);
+            SubscribeLocalEvent<EnergySwordComponent, UseInHandEvent>(OnUseInHand);
+            SubscribeLocalEvent<EnergySwordComponent, InteractUsingEvent>(OnInteractUsing);
+        }
+
+        private void OnComponentInit(EntityUid uid, EnergySwordComponent comp, ComponentInit args)
+        {
+            if (comp.ColorOptions.Count != 0)
+                comp.BladeColor = _random.Pick(comp.ColorOptions);
+        }
+
+        private void OnMeleeHit(EntityUid uid, EnergySwordComponent comp, MeleeHitEvent args)
+        {
+            
+            if (comp.Activated == true)
+            {
+                args.BonusDamage.DamageDict = new Dictionary<string, FixedPoint2>()
+                {
+                    {"Slash", FixedPoint2.New(25)},
+                };
```

Comment:
for now just make it a datafield. I.e:
```cs
args.BonusDamage = comp.LitDamageBonus
```
with 
```cs
[DataField("litDamageBonus", required; true)]
public DamageSpecififer LitDamageBonus = default!;
```
in the component and
```yaml
litDamageBonus:
  types:
    Slash: 12.5
    Burn: 12.5
    Blunt -7 # or whatever the base-sword-bashing damage is
```

Reply:
Done dealio, works wonders.
As always, thanks my dude!